### PR TITLE
Hide enable password login flow until user verification

### PR DESCRIPTION
**Pull request by shishirx34**
Fix for #4984 - Password reset for accounts created with MSA login is broken

Also, added a redundant server side check which would send invalid password reset links. 

use `?w=1` to ignore whitespace changes.

[UPDATE]
Before confirmation:
image

After confirmation:
image

### CODE REVIEW

**Comment by skofman1** [comment]
What will the UI look like for an unconfirmed account? Can you attach a screenshot?

**Comment by shishirx34** [comment]
@skofman1 - see the updated description

File: src/NuGetGallery/Views/Users/Account.cshtml [hunk 1]
```diff
@@ -252,57 +252,64 @@
...
+                                        Cancel
+                                    </a>
+                                </div>
                             </div>
-                        </div>
+                        }
+                    }
+                    else
+                    {
+                        <p>To enable password login, please confirm your account.</p>
```
Comment (chenriksson) [inline src/NuGetGallery/Views/Users/Account.cshtml]: Do we need to provide any more guidance on how to confirm their account?

File: src/NuGetGallery/Controllers/UsersController.cs [hunk 2]
```diff
@@ -575,7 +575,12 @@ public virtual ActionResult Profiles(string username, int page = 1)
             if (oldPassword == null)
             {
                 // User is requesting a password set email
-                await _authService.GeneratePasswordResetToken(user, Constants.PasswordResetTokenExpirationHours * 60);
+                var resetResultType = await _authService.GeneratePasswordResetToken(user, Constants.PasswordResetTokenExpirationHours * 60);
```
Comment (chenriksson) [inline src/NuGetGallery/Controllers/UsersController.cs]: Maybe add a test for this?

File: src/NuGetGallery/Views/Users/Account.cshtml (hunk 1 above)
Comment (scottbommarito) [reply src/NuGetGallery/Views/Users/Account.cshtml]: You can link to the confirmation required page, `account/ConfirmationRequired`

**Pull request merged**